cmd/mountgen1.go

```diff
@@ -0,0 +1,243 @@
+   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+   SOFTWARE
+*/
+
+package cmd
+
+import (
+	"blobfuse2/common"
+	"blobfuse2/common/config"
+	"blobfuse2/common/log"
+	"blobfuse2/component/azstorage"
+	"blobfuse2/component/file_cache"
+	"blobfuse2/component/libfuse"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+var AzStorageOpt azstorage.AzStorageOptions
+var LibFuseOpt libfuse.LibfuseOptions
+var FileCacheOpt file_cache.FileCacheOptions
+var requiredFreeSpace int
+var configFile string
+var generateJsonOnly bool
+var gen1ConfigFilePath string
+
+var gen1Cmd = &cobra.Command{
+	Use:        "mountgen1",
+	Short:      "Mounts Azure Storage ADLS Gen 1 account using SPN auth",
+	Long:       "Mounts Azure Storage ADLS Gen 1 account using SPN auth",
+	SuggestFor: []string{"mntgen1", "gen1 mount"},
+	Args:       cobra.ExactArgs(1),
+	Hidden:     true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		options.MountPath = args[0]
+		options.ConfigFile = configFile
+		parseConfig()
+
+		err := config.Unmarshal(&options)
+		if err != nil {
+			fmt.Printf("Init error config unmarshall [%s]", err)
+			return err
+		}
+
+		err = options.validate(false)
+		if err != nil {
+			fmt.Printf("mountgen1: error invalid options [%v]", err)
+			return err
+		}
+
+		err = config.UnmarshalKey("azstorage", &AzStorageOpt)
+		if err != nil {
+			log.Err("mountgen1: AzStorage config error [invalid config attributes]")
+			return err
+		}
+
+		// not checking ClientSecret since adlsgen1fuse will be reading secret from env variable (ADL_CLIENT_SECRET)
+		if AzStorageOpt.AuthMode == "" && AzStorageOpt.ClientID != "" && AzStorageOpt.TenantID != "" {
+			AzStorageOpt.AuthMode = "spn"
+		}
+
+		err = config.UnmarshalKey("libfuse", &LibFuseOpt)
+		if err != nil {
+			log.Err("mountgen1: Libfuse config error [invalid config attributes]")
+			return err
+		}
+
+		err = config.UnmarshalKey("file_cache", &FileCacheOpt)
+		if err != nil {
+			log.Err("mountgen1: FileCache config error [invalid config attributes]")
+			return err
+		}
+
+		var logLevel common.LogLevel
+		err = logLevel.Parse(options.Logging.LogLevel)
+		if err != nil {
+			fmt.Println("error: invalid log level")
+		}
+
+		err = mountRustFuse()
+		if err != nil {
+			fmt.Printf("Unable to mount Gen1: %s\n", err.Error())
+			return fmt.Errorf("unable to mount Gen1: %s", err.Error())
+		}
+
+		if !generateJsonOnly {
+			err = runAdlsBinary()
+			if err != nil {
+				return fmt.Errorf("unable to run adlsgen1fuse binary: %s", err.Error())
+			}
+		}
+
+		return nil
+	},
+}
+
+// code to generate json file for rustfuse
+func mountRustFuse() error {
```

Comment:
This function is just generating the json config file for gen1fuse so name of the function is not correct